src/parser/mod.rs

```diff
@@ -5723,16 +5734,19 @@ impl<'a> Parser<'a> {
 
         let strict = self.parse_keyword(Keyword::STRICT);
 
-        let comment = if self.parse_keyword(Keyword::COMMENT) {
-            let _ = self.consume_token(&Token::Eq);
-            let next_token = self.next_token();
-            match next_token.token {
-                Token::SingleQuotedString(str) => Some(CommentDef::WithoutEq(str)),
-                _ => self.expected("comment", next_token)?,
-            }
-        } else {
-            None
-        };
+        // For Hive, the comment is after the table column definitions
+        if !dialect_of!(self is HiveDialect) {
```

Comment:
is this correct to switch this to being guarded on the Hive dialect (other dialects didn't support it previously)?

Reply:
I see, the comment line is a bit misleading. I use `!dialect_of!(self is HiveDialect)` here to disallow `Hive` to parse the comment here?

Reply:
Oh right! 😅 my bad seems I'm used to seeing if dialect_of and my brain skipped the !